Scripts/Decorator/ObservingDecorator.cs

```diff
@@ -41,6 +41,18 @@ override protected void DoStop()
             Decoratee.Stop();
         }
 
+        public override void Pause()
+        {
+            base.Pause();
+            StopObserving();
+        }
+
+        public override void Resume()
+        {
+            base.Resume();
+            StartObserving();
```

Comment:
same as for the service, I think it's safer to either call `StartObserving()` before calling `base.Resume()`or to check the Node's current state for being Active when calling `StartObserving()`
(although I'm still not quite sure what's better tbh)